lib/kvdb/ikvdb.c

```diff
@@ -1882,32 +1882,54 @@ kvdb_kvs_set_ikvs(struct kvdb_kvs *kk, struct ikvs *ikvs)
 }
 
 merr_t
-ikvdb_kvs_create(struct ikvdb *handle, const char *kvs_name, const struct kvs_cparams *params)
+ikvdb_kvs_create(struct ikvdb *handle, const char *kvs_name, struct kvs_cparams *params)
 {
+    struct ikvdb_impl *self;
+    struct kvdb_kvs *kvs;
+    merr_t err;
+    int idx;
+
     assert(handle);
     assert(kvs_name);
     assert(params);
 
-    merr_t             err = 0;
-    struct ikvdb_impl *self = ikvdb_h2r(handle);
-    struct kvdb_kvs *  kvs = NULL;
-    int                idx;
-
+    self = ikvdb_h2r(handle);
     if (self->ikdb_read_only)
-        goto out_immediate;
+        return 0;
```

Comment:
return error here?

Reply:
It didn't used to return an error, so I don't want to break it..  Dunno..

Reply:
bizarre

Reply:
we should return an error here for 3.0